base/CMakeLists.txt

```diff
@@ -73,22 +73,21 @@ foreach(dir ${OSX_EXTRA_LIBRARY_PATH})
   target_link_libraries(${this} PUBLIC "-Xlinker -rpath -Xlinker ${dir}")
 endforeach()
 
-ecbuild_add_executable (TARGET cub2latlon.x SOURCES cub2latlon_regridder.F90)
-target_link_libraries (cub2latlon.x ${this} MAPL.pfio ${ESMF_LIBRARIES} ${MPI_Fortran_LIBRARIES} ${OpenMP_Fortran_LIBRARIES})
+ecbuild_add_executable (TARGET cub2latlon.x SOURCES cub2latlon_regridder.F90 DEPENDS esmf)
+target_link_libraries (cub2latlon.x ${this} MAPL.pfio ${MPI_Fortran_LIBRARIES} ${OpenMP_Fortran_LIBRARIES})
 set_target_properties (cub2latlon.x PROPERTIES Fortran_MODULE_DIRECTORY ${esma_include}/${this} LINK_FLAGS "${OpenMP_Fortran_FLAGS}")
 
-ecbuild_add_executable (TARGET Regrid_Util.x SOURCES Regrid_Util.F90)
-target_link_libraries (Regrid_Util.x ${this} MAPL.pfio ${ESMF_LIBRARIES} ${MPI_Fortran_LIBRARIES} ${OpenMP_Fortran_LIBRARIES})
+ecbuild_add_executable (TARGET Regrid_Util.x SOURCES Regrid_Util.F90 DEPENDS esmf)
+target_link_libraries (Regrid_Util.x ${this} MAPL.pfio ${MPI_Fortran_LIBRARIES} ${OpenMP_Fortran_LIBRARIES})
 set_target_properties (Regrid_Util.x PROPERTIES Fortran_MODULE_DIRECTORY ${esma_include}/${this} LINK_FLAGS "${OpenMP_Fortran_FLAGS}")
 
 if (EXTENDED_SOURCE)
-#    target_compile_options (${this} PRIVATE $<$<COMPILE_LANGUAGE:Fortran>:${EXTENDED_SOURCE}>)
-esma_fortran_generator_list (${this} ${EXTENDED_SOURCE})
+  esma_fortran_generator_list (${this} ${EXTENDED_SOURCE})
 endif()
 
 #--------------------
-target_include_directories (${this} PUBLIC ${INC_ESMF} ${INC_NETCDF})
-target_link_libraries (${this} PUBLIC ${ESMF_LIBRARIES} ${MPI_Fortran_LIBRARIES})
+add_dependencies (${this} esmf NetCDF::NetCDF_Fortran MPI::MPI_Fortran)
```

Comment:
`target_link_libraries` ?

Reply:
@tclune, are you suggesting to remove `add_dependencies()`?

Reply:
We may remove both `add_dependencies()` and `target_link_libraries()` and apply the following code change:
```
--- a/base/CMakeLists.txt
+++ b/base/CMakeLists.txt
@@ -52,7 +52,8 @@ endif()
 
 esma_add_library(
   ${this} SRCS ${srcs}
-  DEPENDENCIES MAPL.shared MAPL.profiler MAPL.pfio MAPL_cfio_r4 gftl-shared pflogger MPI::MPI_Fortran)
+  DEPENDENCIES MAPL.shared MAPL.profiler MAPL.pfio MAPL_cfio_r4 pflogger gftl-shared
+               esmf NetCDF::NetCDF_Fortran MPI::MPI_Fortran)
 
 target_compile_options (${this} PRIVATE $<$<COMPILE_LANGUAGE:Fortran>:${OpenMP_Fortran_FLAGS}>)
 if(DISABLE_GLOBAL_NAME_WARNING)
@@ -86,8 +87,8 @@ if (EXTENDED_SOURCE)
 endif()
 
 #--------------------
-add_dependencies (${this} esmf NetCDF::NetCDF_Fortran MPI::MPI_Fortran)
-target_link_libraries (${this} PUBLIC esmf NetCDF::NetCDF_Fortran MPI::MPI_Fortran)
 
 # Users guide 
 if (LATEX_FOUND)
```

Reply:
Sounds good to me.   The reason it was the other way is that this code preceded the use of  `find_package()` for NetCDF.   This is a good time to fix.

Reply:
Implemented in fb2f35a